src/data/services/LmsApiService.js

```diff
@@ -235,12 +237,30 @@ class LmsApiService {
     return LmsApiService.apiClient().get(`${LmsApiService.lmsIntegrationUrl}/configs/?${queryParams.toString()}`);
   }
 
-  static fetchContentMetadataItemTransmission(uuid, channelCode, configId) {
-    return LmsApiService.apiClient().get(`${LmsApiService.lmsContentSyncStatusUrl}/${uuid}/${channelCode}/${configId}`);
+  static generateFormattedStatusUrl(url, currentPage, options) {
+    // pages index from 1 in backend, frontend components index from 0
+    let pageToUse;
+    if (currentPage) {
+      pageToUse = currentPage + 1;
+    }
+    let queryParams = '';
+    if (!_.isEmpty(options)) {
+      queryParams = new URLSearchParams(options);
+    }
+    const paramString = `?${pageToUse ? `page=${pageToUse}` : ''}&${queryParams.toString()}`;
+    return `${url}${paramString}`;
+  }
+
+  static fetchContentMetadataItemTransmission(uuid, channelCode, configId, currentPage, options) {
+    const syncStatusPath = `${LmsApiService.lmsContentSyncStatusUrl}/${uuid}/${channelCode}/${configId}`;
+    const getSyncStatusUrl = LmsApiService.generateFormattedStatusUrl(syncStatusPath, currentPage, options);
+    return LmsApiService.apiClient().get(getSyncStatusUrl);
   }
 
-  static fetchLearnerMetadataItemTransmission(uuid, channelCode, configId) {
-    return LmsApiService.apiClient().get(`${LmsApiService.lmsLearnerSyncStatusUrl}/${uuid}/${channelCode}/${configId}`);
+  static fetchLearnerMetadataItemTransmission(uuid, channelCode, configId, currentPage, options) {
+    const syncStatusPath = `${LmsApiService.lmsLearnerSyncStatusUrl}/${uuid}/${channelCode}/${configId}`;
+    const getSyncStatusUrl = LmsApiService.generateFormattedStatusUrl(syncStatusPath, currentPage, options);
+    return LmsApiService.apiClient().get(getSyncStatusUrl);
   }
 
   /**
```

Comment:
I know that you're doing more checks on the backend, but maybe just adding a basic test on traversing through multiple pages on the frontend? Just as a sanity check

Reply:
done! 😄